From 85a1e83c5b2174b2f3da38d7d5517f8f2281b7e0 Mon Sep 17 00:00:00 2001
From: Lukasz Bartosik <lukasz.bartosik@cavium.com>
Date: Wed, 30 Jan 2019 11:43:11 +0100
Subject: [PATCH 0938/1051] crypto: cpt - add write barrier before LMTST

Add write memory barrier before LMTST operation to CPT.

Change-Id: I96751e97090b3416da60f90cdd986d5ebb9487c4
Signed-off-by: Lukasz Bartosik <lukasz.bartosik@cavium.com>
Reviewed-on: https://sj1git1.cavium.com/3023
Reviewed-by: Srujana Challa <schalla@cavium.com>
Tested-by: sa_ip-sw-jenkins
[Kevin: The original patch got from Marvell sdk10.0_19.06]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/crypto/cavium/cpt/9x/cpt9x_reqmgr.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/crypto/cavium/cpt/9x/cpt9x_reqmgr.c b/drivers/crypto/cavium/cpt/9x/cpt9x_reqmgr.c
index 0aad4abef403..17083b9d7ca8 100644
--- a/drivers/crypto/cavium/cpt/9x/cpt9x_reqmgr.c
+++ b/drivers/crypto/cavium/cpt/9x/cpt9x_reqmgr.c
@@ -105,6 +105,12 @@ void cpt9x_send_cmd(union cpt_inst_s *cptinst, u32 insts_num, void *obj)
 	void *ioreg = lf->ioreg;
 	long ret;
 
+	/*
+	 * Make sure memory areas pointed in CPT_INST_S
+	 * are flushed before the instruction is sent to CPT
+	 */
+	smp_wmb();
+
 	do {
 		/* Copy CPT command to LMTLINE */
 		memcpy(lmtline, cptinst, insts_num * CPT_INST_SIZE);
-- 
2.17.1

